includes/core/classes/class-event-feed.php

```diff
@@ -0,0 +1,321 @@
+	protected function __construct() {
+		// Initialize the rewrite slug from settings.
+		$settings           = Settings::get_instance();
+		$this->rewrite_slug = $settings->get_value( 'general', 'urls', 'events' );
+
+		$this->setup_hooks();
+	}
+
+	/**
+	 * Set up hooks for various purposes.
+	 *
+	 * This method adds hooks for different purposes as needed.
+	 *
+	 * @since 1.0.0
+	 *
+	 * @return void
+	 */
+	protected function setup_hooks(): void {
+		// Add hooks for flexible event feed customization.
+		add_filter( 'gatherpress_event_feed_excerpt', array( $this, 'get_default_event_excerpt' ) );
+		add_filter( 'gatherpress_event_feed_content', array( $this, 'get_default_event_content' ) );
+
+		// Apply feed customization only if no theme/editor overrides.
+		add_filter( 'the_excerpt_rss', array( $this, 'apply_event_excerpt' ) );
+		add_filter( 'the_content_feed', array( $this, 'apply_event_content' ) );
+
+		// Hook into the main query to handle events feeds.
+		add_action( 'pre_get_posts', array( $this, 'handle_events_feed_query' ) );
+	}
+
+	/**
+	 * Handle events feed queries by setting the appropriate parameters.
+	 *
+	 * This method works in conjunction with Event_Query::prepare_event_query_before_execution()
+	 * to ensure feeds show upcoming events with proper sorting.
+	 *
+	 * @since 1.0.0
+	 *
+	 * @param WP_Query $query The WP_Query instance.
+	 * @return void
+	 */
+	public function handle_events_feed_query( WP_Query $query ): void {
+		// Only run on the main query and if it's a feed.
+		if ( ! $query->is_main_query() || ! is_feed() ) {
+			return;
+		}
+
+		// Check if this is an events feed request.
+		if ( isset( $_SERVER['REQUEST_URI'] ) ) {
+			$request_uri = sanitize_text_field( wp_unslash( $_SERVER['REQUEST_URI'] ) );
+
+			// Check if this is the events feed URL.
+			if ( str_contains( $request_uri, '/' . $this->rewrite_slug . '/' . $GLOBALS['wp_rewrite']->feed_base ) ) {
+				// Set the post type and let Event_Query handle the rest.
+				$query->set( 'post_type', Event::POST_TYPE );
+				$query->set( 'gatherpress_events_query', 'upcoming' );
+			}
+		}
+	}
+
+	/**
+	 * Get formatted event datetime information for feeds.
+	 *
+	 * @since 1.0.0
+	 *
+	 * @param Event $event The event object.
+	 * @return array Array of event information strings.
+	 */
+	private function get_event_datetime_info( Event $event ): array {
+		$event_info       = array();
+		$display_datetime = $event->get_display_datetime();
+
+		if ( ! empty( $display_datetime ) && '—' !== $display_datetime ) {
+			$event_info[] = sprintf(
+				/* translators: %s: Formatted date and time */
+				__( 'Date: %s', 'gatherpress' ),
+				$display_datetime
+			);
+		}
+
+		return $event_info;
+	}
+
+	/**
+	 * Get default event excerpt customization.
+	 *
+	 * @since 1.0.0
+	 *
+	 * @param string $excerpt The current excerpt.
+	 * @return string The customized excerpt.
+	 */
+	public function get_default_event_excerpt( string $excerpt ): string {
+		global $post;
```

Comment:
```suggestion
		if ( Event::POST_TYPE !== get_post_type() ) {
			return $excerpt;
		}
```

I already mentioned this before, but I guess you may have missed this @jmarx 

I would remove the `global $post `and rely on `get_post_type()` falling back to the global internaly.

Reply:
@carstingaxion Fixed. I swear I tried this and it was not working before. But it's good now.